EasyReflectometry/sample/assemblies/base_assembly.py

```diff
@@ -11,6 +11,10 @@
 
 
 class BaseAssembly(BaseObj):
+    """Assembly of layers.
+    The front layer faces the neutron beam and has index 0.
```

Comment:
For a `sample` the front (super phase) and back (sub phase) would be semi infinite layers.  A `base_assemply` will be padded with other layers and the `front` and `back` here serve to make it clear where the neutron beam enters and exists.

Changed to:
```
    The front layer (front_layer) is the layer the neutron beam starts in, it has an index of 0.
    The back layer (back_layer) is the final layer from which the unreflected neutron beam is transmitted,
    its index number depends on the number of finite layers in the system, but it might be accessed at index -1.
```